scripts/downloads/download_data.py

```diff
@@ -140,43 +227,49 @@ def main(
         which capture to download.
     - blender: Blender synthetic scenes realeased with NeRF.
     - friends: Friends TV show scenes.
+    - record3d: Record3d dataset.
+    - dnerf: D-NeRF dataset.
+    - phototourism: PhotoTourism dataset. Use the `capture_name` argument to specify which capture to download.
 
     Args:
         dataset: The dataset to download (from).
-        capture_name: The capture name to download (if downloading from nerfstudio dataset).
-        save_dir: The directory to save the data to.
     """
     save_dir.mkdir(parents=True, exist_ok=True)
 
-    if dataset == "blender":
-        if capture_name is not None:
-            CONSOLE.print("Capture name is ignored when downloading from the blender dataset.")
-        download_blender(save_dir)
-    if dataset == "friends":
-        if capture_name is not None:
-            CONSOLE.print("Capture name is ignored when downloading from the blender dataset.")
-        download_friends(save_dir)
-    if dataset == "nerfstudio":
-        if capture_name is None:
-            capture_names = sorted(nerfstudio_file_ids.keys())
-            CONSOLE.rule("[bold red]Error", style="bold red")
-            CONSOLE.print("[bold yellow]You must pass in --capture-name when downloading from the nerfstudio dataset.")
-            CONSOLE.print("Use one of the following:")
-            CONSOLE.print(f"\t {capture_names}")
-            sys.exit()
-        download_nerfstudio(save_dir, capture_name)
-    if dataset == "record3d":
-        download_record3d(save_dir, "bear")
-    if dataset == "dnerf":
-        if capture_name is not None:
-            CONSOLE.print("Capture name is ignored when downloading from the dnerf dataset.")
-        download_dnerf(save_dir)
+    dataset.download(save_dir)
+
+
+method_configs = {
+    "blender": BlenderDownload(),
+    "friends": FriendsDownload(),
+    "nerfstudio": NerfstudioDownload(),
+    "record3d": Record3dDownload(),
+    "dnerf": DNerfDownload(),
+    "phototourism": PhototourismDownload(),
+}
```

Comment:
Do something like this instead. It's quite a bit cleaner. https://github.com/nerfstudio-project/nerfstudio/blob/009e21be248418028e6303ecde593a20ef88fa87/scripts/process_data.py#L1174-L1180